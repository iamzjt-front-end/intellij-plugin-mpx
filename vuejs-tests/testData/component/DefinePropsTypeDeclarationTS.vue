<script setup lang="ts">
defineProps<{
  optional?: string,
  required: string,
  requiredNullish: string|undefined,
}>();

optional; // TODO Should be unresolved
</script>

<template>
  <<warning descr="Element DefinePropsTypeDeclarationTS doesn't have required attribute required-nullish">DefinePropsTypeDeclarationTS</warning>
    v-if="false"
    :required="'something'"
    :optional="'something'"
  />
  {{optional}}
  {{required}}
  {{requiredNullish}}
</template>
