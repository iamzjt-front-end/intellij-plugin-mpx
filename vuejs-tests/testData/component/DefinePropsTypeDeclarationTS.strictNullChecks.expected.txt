{
 class: "VueSourceComponent",
 components: {
  DefinePropsTypeDeclarationTS: <recursive value of class VueSourceComponent>,
 },
 props: [
  {
   class: "VueScriptSetupInputProperty",
   name: "optional",
   jsType: "string | undefined",
   required: false,
  },
  {
   class: "VueScriptSetupInputProperty",
   name: "required",
   jsType: "string",
   required: true,
  },
  {
   class: "VueScriptSetupInputProperty",
   name: "requiredNullish",
   jsType: "string | undefined",
   required: true,
  },
 ],
}
